transformer_engine/pytorch/csrc/comm_gemm_overlap.h

```diff
@@ -20,6 +20,7 @@
 #include "common/util/logging.h"
 #include "common/util/system.h"
 #include "userbuffers/userbuffers.h"
+#include "extensions.h"
```

Comment:
This include is causing build errors since [extensions.h](https://github.com/NVIDIA/TransformerEngine/blob/5143dcf36c7f9af1576fda45c346303acf17f441/transformer_engine/pytorch/csrc/extensions.h) does not have an include guards. We should add a `#pragma once` in extensions.h and comm_gemm_overlap.h (the other headers are properly guarded).